ehcache-107/src/main/java/org/ehcache/jsr107/Eh107CacheLoaderWriterProvider.java

```diff
@@ -15,21 +15,30 @@
  */
 package org.ehcache.jsr107;
 
+import java.util.HashSet;
+import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
+import org.ehcache.core.spi.ServiceLocator;
 import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
-import org.ehcache.impl.internal.spi.loaderwriter.DefaultCacheLoaderWriterProvider;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriterConfiguration;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriterProvider;
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceProvider;
 
 /**
  * @author teck
  */
-class Eh107CacheLoaderWriterProvider extends DefaultCacheLoaderWriterProvider {
+class Eh107CacheLoaderWriterProvider implements CacheLoaderWriterProvider {
```

Comment:
We're attempting here to decouple this class from the default cache loader writer provider. This eliminates the need to export an internal package from the implementation OSGi bundle.